test/test_lyric.py

```diff
@@ -0,0 +1,397 @@
+        lyrics = create_lyrics()
+        lyrics.create_line(10)
+        lyrics.create_line(20)
+        index_generator = lyrics.get_index_of_line_at_tick(10)
+        self.assertEqual(next(index_generator), 1)
+        self.assertEqual(next(index_generator), 0)
+
+    def test_remove_word_at_index(self):
+        lyrics = create_lyrics()
+        line = lyrics.create_line(9)
+        line.create_word("test", 15, 20)
+        line.create_word("word", 20, 30)
+        # Replace the empty word
+        self.assertEqual(len(line), 2)
+        assert_lyric_words_equal({
+            "word": "test",
+            "start_tick": 15,
+            "end_tick": 20,
+        }, line[0])
+        assert_lyric_words_equal({
+            "word": "word",
+            "start_tick": 20,
+            "end_tick": 30,
+        }, line[1])
+        line.remove_word_at_index(0)
+        # There will be 1 placeholder left
+        self.assertEqual(len(line), 1)
+        assert_lyric_words_equal({
+            "word": "word",
+            "start_tick": 20,
+            "end_tick": 30,
+        }, line[0])
+        # Test changed order of lines
+        assert_lyric_lines_equal({
+            "sentence": "word",
+            "start_tick": 20,
+            "end_tick": 30,
+        }, lyrics[1])
+
+    def test_remove_word(self):
+        lyrics = create_lyrics()
+        line = lyrics.create_line(10)
+        word = line.create_word("first", 15, 20)
+        line.create_word("second", 30, 50)
+        assert_lyric_words_equal({"word": "first", "start_tick": 15, "end_tick": 20}, line[0])
+        line.remove_word(word)
+        assert_lyric_words_equal({"word": "second", "start_tick": 30, "end_tick": 50}, line[0])
+        assert_lyric_lines_equal({
+            "sentence": "second",
+            "start_tick": 30,
+            "end_tick": 50,
+        }, lyrics[1])
+        self.assertEqual(len(line), 1)
+
+
+class TestLyricWord(unittest.TestCase):
+    def test_set_word(self):
+        lyrics = create_lyrics()
+        line = lyrics.create_line(10)
+        word = line[0]
+        word = LyricWord(line, "test", 10, 20, word)
+        word.set_word("changed")
+        self.assertEqual(word.get_word(), "changed")
+
+    def test_set_start_tick(self):
+        lyrics = create_lyrics()
+        line = lyrics.create_line(10)
+        word = line.create_word("test", 15, 20)
+        line.create_word("tick", 30, 50)
+        assert_lyric_lines_equal({
+            "sentence": "testtick",
+            "start_tick": 15,
+            "end_tick": 50,
+        }, lyrics[1])
+        word.set_start_tick(5)
+        self.assertEqual(word.get_start_tick(), 5)
+        self.assertEqual(line.get_start_tick(), 5)
+        # Test if lines are sorted
+        assert_lyric_lines_equal({
+            "sentence": "testtick",
+            "start_tick": 5,
+            "end_tick": 50,
+        }, lyrics[0])
+        # Test removing the updated word
+        lyrics[0].remove_word(word)
```

Comment:
I meant setting a start tick or end tick that could delete the word, for example here if you set start tick to be >= end tick we should delete the word from the line. This is not tested currently.